src/crystal/system/win32/path.cr

```diff
@@ -0,0 +1,17 @@
+require "c/combaseapi"
+require "c/knownfolders"
+require "c/shlobj_core"
+
+module Crystal::System::Path
+  def self.home : String
+    if home_path = ENV["USERPROFILE"]?
+      home_path
+    elsif LibC.SHGetKnownFolderPath(pointerof(LibC::FOLDERID_Profile), 0, nil, out path_ptr) == 0
+      home_path = String.from_utf16(path_ptr)[0]
```

Comment:
I think tuple destructuring makes it easier to follow. It's easy to assume `String.from_utf16` returns `String` - there is an overload that does - and then you'd be confused why to pick only the first character from that string.
```suggestion
      home_path, _ = String.from_utf16(path_ptr)
```